namui/sample/image-load-background/src/lib.rs

```diff
@@ -9,52 +9,77 @@ pub fn main() {
-        let y_index_length = (image_urls.len() as f32 / x_index_length as f32).ceil() as usize;
+        let y_index_length =
+            (image_resource_locations.len() as f32 / x_index_length as f32).ceil() as usize;
         let image_width = size.width.into_px() / x_index_length as f32;
         let image_height = size.height.into_px() / y_index_length as f32;
 
         ctx.compose(|ctx| {
             for x in 0..x_index_length {
                 for y in 0..y_index_length {
-                    if let Some(image_url) = image_urls.get(x + y * 6) {
+                    ctx.compose(|ctx| {
+                        let Some(resource_location) = image_resource_locations.get(x + y * 6)
+                        else {
+                            return;
+                        };
                         let key = format!("{}-{}", x, y);
+                        let rect = Rect::Xywh {
+                            x: image_width * x,
+                            y: image_height * y,
+                            width: image_width,
+                            height: image_height,
+                        };
                         ctx.add_with_key(
                             key,
-                            namui::image(ImageParam {
-                                rect: Rect::Xywh {
-                                    x: image_width * x,
-                                    y: image_height * y,
-                                    width: image_width,
-                                    height: image_height,
-                                },
-                                source: ImageSource::Url {
-                                    url: image_url.clone(),
-                                },
-                                style: ImageStyle {
-                                    fit: ImageFit::Contain,
-                                    paint: None,
-                                },
-                            }),
+                            SampleImage {
+                                resource_location,
+                                rect,
+                            },
                         );
-                    }
+                    });
                 }
             }
         });
+    }
+}
 
-        
+struct SampleImage<'a> {
+    resource_location: &'a ResourceLocation,
+    rect: Rect<Px>,
+}
+impl Component for SampleImage<'_> {
+    fn render(self, ctx: &RenderCtx) {
+        let Self {
+            resource_location,
+            rect,
+        } = self;
+        let image = ctx.image(resource_location);
+        ctx.compose(|ctx| {
+            let Some(Ok(image)) = image.as_ref() else {
+                return;
+            };
+            ctx.add(namui::image(ImageParam {
+                rect,
+                image: image.clone(),
+                style: ImageStyle {
+                    fit: ImageFit::Contain,
+                    paint: None,
+                },
+            }));
+        });
```

Comment:
I think this should be the default behavior supported by namui.
Most of cases, user don't want to take care whether the image is loaded or not, and they just want to put src for namui to draw image.

Please consider below cases
1. Manually load image, get image info, then render.
2. Just put image source to namui, Don't care whether image is loaded or not exists, like web's img.

The second option may break the perfectness, But I don't want user to check image's loading and existence every time, especially when the image is from bundle, which we can make sure it should be loaded.

Reply:
Then we can do the followings

1. Keep `ImageDrawCommand` for case 1
2. Change `namui::image()` to receive just source for case 2

Reply:
I guess there is `ImageRender` https://github.com/NamseEnt/namseent/pull/872/files#diff-b4dc7e7c2474ffadea70322783d4d25589cdfc21a17526dca895db2ddb11b3f8R31 so I don't think we need to provide another wrapper?

Reply:
If ImageRender can cover that case, could you use it and make the sample simple? Because sample is the kind of manual of namui, so I hope it should be simple as we want it to be.

Reply:
I didn't know that there is `ImageRender`
sample updated